File: pipeline/kite9-pipeline-common/src/main/java/com/kite9/k9server/command/xml/replace/ReplaceStyle.java
package com.kite9.k9server.command.xml.replace;

import org.kite9.diagram.dom.elements.StyledKite9XMLElement;
import org.springframework.util.StringUtils;
import org.w3c.dom.Element;
import org.w3c.dom.css.CSSStyleDeclaration;

import com.kite9.k9server.adl.holder.pipeline.ADLDom;

public class ReplaceStyle extends AbstractReplaceCommand<String, String> {

	public String name;
	
	@Override
	protected String getFromContent(ADLDom context) {
		return from;
	}

	@Override
	protected String getToContent(ADLDom context) {
		return to;
	}

	@Override
	protected String getExistingContent(ADLDom in) {
		Element e = findFragmentElement(in.getDocument(), fragmentId);
		if (e instanceof StyledKite9XMLElement) {
			CSSStyleDeclaration sd = ((StyledKite9XMLElement) e).getStyle();
			return sd.getPropertyValue(name);
		} else {
			return "";
		}
	}

	@Override
	protected void doReplace(ADLDom in, String site, String toContent, String fromContent) {
		Element e = findFragmentElement(in.getDocument(), fragmentId);
		if (e instanceof StyledKite9XMLElement) {
			CSSStyleDeclaration sd = ((StyledKite9XMLElement) e).getStyle();
			if (toContent == null) {
				sd.removeProperty(name);
			} else {
				sd.setProperty(name, toContent, "");
			}
			
			if (sd.getLength() == 0) {
				((StyledKite9XMLElement) e).removeAttribute("style");
			}
		}
	}
	
	@Override
	protected void checkProperties() {
		ensureNotNull("name", name);
		super.checkProperties();
	}

	@Override
	protected Mismatch same(String existing, String with) {
		if (StringUtils.isEmpty(existing) && StringUtils.isEmpty(with)) {
			return null;
		}
		return super.same(existing, with);
	}
	
	
}

File: pipeline/kite9-pipeline-common/src/main/kotlin/com/kite9/pipeline/command/xml/replace/ReplaceStyle.kt
package com.kite9.pipeline.command.xml.replace

import com.kite9.pipeline.adl.holder.pipeline.ADLDom
import com.kite9.pipeline.command.Command
import com.kite9.pipeline.command.CommandContext

class ReplaceStyle : AbstractReplaceCommand<String?, String?>() {

    @JvmField
    var name: String = ""

    override fun getFromContent(context: ADLDom, ctx: CommandContext): String? {
        return from
    }

    override fun getToContent(context: ADLDom, ctx: CommandContext): String? {
        return to
    }

    override fun getExistingContent(o: ADLDom, ctx: CommandContext): String? {
        val e = findFragmentElement(o.document, fragmentId, ctx)
        return ctx.getStyleValue(e, name)
    }

    override fun doReplace(
        on: ADLDom,
        site: String?,
        toContent: String?,
        fromContent: String?,
        ctx: CommandContext
    ) : Command.Mismatch? {
        val e = findFragmentElement(on.document, fragmentId, ctx)

        if (e==null) {
            return Command.Mismatch { "Couldn't find element with id "+fragmentId }
        }

        ctx.setStyleValue(e, name, toContent)

        return null
    }

    override fun checkProperties() {
        ensureNotNull("name", name)
        super.checkProperties()
    }

    override fun same(existing: String?, with: String?, ctx: CommandContext): Command.Mismatch? {
        return if (isEmpty(existing) && isEmpty(with)) {
            null
        } else super.same(existing, with, ctx)
    }
}